### PR TITLE
Fix fields from other ITIL objects during creation

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #12182

### CODE REVIEW

**Comment by cconard96** [comment]
The initForm call was moved just after the setting of the fields from another ITIL Object. While this works, it looks like the predefined values from templates override what is inherited. I don't think that is desired is it?

**Review by orthagh — DISMISSED** [review]

**Review by trasher — APPROVED** [review]

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -497,6 +495,8 @@ protected function setPredefinedFields(ITILTemplate $tt, array &$options, array
             $options['entities_id']         = $problem->fields['entities_id'];
         }
 
+        $this->initForm($this->fields['id'], $options);
```
Comment (cedric-anne) [inline src/CommonITILObject.php:498]: Calling `initForm` in `setPredefinedFields` seems weird to me.

Comment (orthagh) [reply src/CommonITILObject.php:498]: oh i missed that. Indeed, initForm should be kept in showForm function

**Review by cedric-anne — APPROVED** [review]
Seems OK, but I hope there is no side effect.

**Comment by trasher** [comment]
Please rebase to solve conflict

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**